src/test/java/com/iexec/sms/secret/SecretControllerTests.java

```diff
@@ -0,0 +1,318 @@
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.*;
+
+class SecretControllerTests {
+
+    private static final String AUTHORIZATION = "AUTHORIZATION";
+    private static final String CHALLENGE = "CHALLENGE";
+    private static final String WEB2_OWNER_ADDRESS = "WEB2_OWNER_ADDRESS";
+    private static final String WEB2_SECRET_NAME = "WEB2_SECRET_NAME";
+    private static final String WEB2_SECRET_VALUE = "WEB2_SECRET_VALUE";
+    private static final String WEB3_SECRET_ADDRESS = "WEB3_SECRET_ADDRESS";
+    private static final String WEB3_SECRET_VALUE = "WEB3_SECRET_VALUE";
+
+    @Mock
+    private AuthorizationService authorizationService;
+
+    @Mock
+    private Web2SecretsService web2SecretsService;
+
+    @Mock
+    private Web3SecretService web3SecretService;
+
+    @InjectMocks
+    private SecretController secretController;
+
+    private static final SecureRandom seed = new SecureRandom();
+
+    @BeforeEach
+    public void init() {
+        MockitoAnnotations.openMocks(this);
+    }
+
+    //region isWeb3SecretSet
+    @Test
+    void shouldReturnNoContentWhenWeb3SecretExists() {
+        when(web3SecretService.getSecret(WEB3_SECRET_ADDRESS))
+                .thenReturn(Optional.of(new Web3Secret()));
+        assertThat(secretController.isWeb3SecretSet(WEB3_SECRET_ADDRESS))
+                .isEqualTo(ResponseEntity.noContent().build());
+        verifyNoInteractions(authorizationService, web2SecretsService);
+    }
+
+    @Test
+    void shouldReturnNotFoundWhenWeb3SecretDoesNotExist() {
+        when(web3SecretService.getSecret(WEB3_SECRET_ADDRESS))
+                .thenReturn(Optional.empty());
+        assertThat(secretController.isWeb3SecretSet(WEB3_SECRET_ADDRESS))
+                .isEqualTo(ResponseEntity.notFound().build());
+        verifyNoInteractions(authorizationService, web2SecretsService);
+    }
+    //endregion
+
+    //region getWeb3Secret
+    @Test
+    void failToGetWeb3SecretWhenBadAuthorization() {
+        when(authorizationService.getChallengeForGetWeb3Secret(WEB3_SECRET_ADDRESS))
+                .thenReturn(CHALLENGE);
+        when(authorizationService.isSignedByOwner(CHALLENGE, AUTHORIZATION, WEB3_SECRET_ADDRESS))
+                .thenReturn(false);
+        assertThat(secretController.getWeb3Secret(AUTHORIZATION, WEB3_SECRET_ADDRESS, false))
+                .isEqualTo(ResponseEntity.status(HttpStatus.UNAUTHORIZED).build());
+        verifyNoInteractions(web2SecretsService, web3SecretService);
+    }
+
+    @Test
+    void failToGetWeb3SecretWhenSecretDoesNotExist() {
+        when(authorizationService.getChallengeForGetWeb3Secret(WEB3_SECRET_ADDRESS))
+                .thenReturn(CHALLENGE);
+        when(authorizationService.isSignedByOwner(CHALLENGE, AUTHORIZATION, WEB3_SECRET_ADDRESS))
+                .thenReturn(true);
+        when(web3SecretService.getSecret(WEB3_SECRET_ADDRESS)).thenReturn(Optional.empty());
```

Comment:
```suggestion
        when(web3SecretService.getSecret(WEB3_SECRET_ADDRESS, false)).thenReturn(Optional.empty());
```

As per my understanding, we don't use `web3SecretService.getSecret(String)`. We use `web3SecretService.getSecret(String, boolean)` instead. I guess this test success because the default return value is `Optional.empty()`.

Reply:
Thanks for your awareness, I'm not sure if mocked tests are what we need for those tests.
It is to easy to do a mistake.
Fixed in 60a8575

Reply:
Fixed in 5a1d5f6